etlman/projects/views.py

```diff
@@ -76,3 +75,135 @@ def new_project_view(request):
+def pipeline_list(request, project_id):
+    project = get_object_or_404(Project, pk=project_id)
+    pipelines = Pipeline.objects.filter(project=project)
+    context = {"pipeline_list": pipelines, "current_project": project}
+    return render(request, "projects/pipeline_list.html", context)
+
+
+@authorize(user_is_project_collaborator)
+def new_pipeline_step1(request, project_id):
+    project = get_object_or_404(Project, pk=project_id)
+    username = request.user.username
+    # Form functionality
+    if request.method == "POST":
+        form_pipeline = PipelineForm(request.POST)
+        form_datainterface = DataInterfaceForm(request.POST)
+        if form_pipeline.is_valid() and form_datainterface.is_valid():
+
+            filled_datainterface = form_datainterface.save(commit=False)
+            filled_datainterface.name = request.POST.getlist("name")[-1]
+            request.session["data_interface"] = model_to_dict(filled_datainterface)
+
+            filled_pipeline = form_pipeline.save(commit=False)
+            filled_pipeline.name = request.POST.getlist("name")[0]
+            filled_pipeline.input_id = filled_datainterface.pk
+            request.session["pipeline"] = model_to_dict(filled_pipeline)
+
+            return HttpResponseRedirect(
+                reverse("projects:new_step", args=(project.pk,))
+            )
+    else:  # GET
+        # If "in_transaction" is on session, the user came back from multiform step2
+        if request.session.get("in_transaction", False):
+            form_pipeline = PipelineForm(initial=request.session["pipeline"])
+            form_datainterface = DataInterfaceForm(
+                initial=request.session["data_interface"]
+            )
+        else:
+            clear_session(request)
+            form_pipeline, form_datainterface = PipelineForm(), DataInterfaceForm()
+
+    context = {
+        "form_pipeline": form_pipeline,
+        "form_datainterface": form_datainterface,
+        "username": username,
+        "current_project": project,
+    }
+    request.session["in_transaction"] = False
+    return render(request, "projects/new_pipeline.html", context)
+
+
+@authorize(user_is_project_collaborator)
+def new_step_step2(request, project_id, step_id=None):
+    project = get_object_or_404(Project, pk=project_id)
+    step = get_object_or_404(Step, pk=step_id) if step_id else None
+    form_data_interface = request.session["data_interface"]
+    form_pipeline = request.session["pipeline"]
+
+    # Form functionality
+    if request.method == "POST":
+        form_step = NewStepForm(request.POST, instance=step)
+        if "save" in request.POST and form_step.is_valid():
+
+            # New Data Interface
+            saved_data_interface = DataInterface.objects.create(
+                # # project=project,
+                # {**form_data_interface},
+                project=project,  # Project(*form_data_interface)
+                name=form_data_interface["name"],
```

Comment:
Shorten this as discussed

Reply:
We tried multiple variations of collection destruction here, but because the end result has more fields than what the form is expecting it, it errors out.

To be specific, it has "id" as extra.

So is it still worth it to go `{k:v for k, v in **form_data_interface if k is not 'id"}` or something similar ?